app/models/agents/email_digest_agent.rb

```diff
@@ -29,22 +33,20 @@ def default_options
 
     def receive(incoming_events)
       incoming_events.each do |event|
-        self.memory['queue'] ||= []
-        self.memory['queue'] << event.payload
```

Comment:
Agreed about the validation.

Instead of a migration, you could probably just add 

```ruby
self.memory.delete('queue') # remove the old queue, we use only 'events' now
```

and we can remove it eventually?  I guess that's a bit hacky, but it's pretty harmless.